wallet/src/main/java/com/android/identity_credential/wallet/presentation/OpenID4VPPresentationActivity.kt

```diff
@@ -479,7 +484,24 @@ class OpenID4VPPresentationActivity : FragmentActivity() {
         //  https://identity.foundation/presentation-exchange/spec/v2.0.0/#input-descriptor
         //
         val inputDescriptorObj = inputDescriptors[0].jsonObject
-        val docType = inputDescriptorObj["id"]!!.toString().run { substring(1, this.length - 1) }
+        val docType = if (credentialFormat == CredentialFormat.MDOC_MSO) {
+            inputDescriptorObj["id"]!!.toString().run { substring(1, this.length - 1) }
+        } else {
+            try {
+                var vct = ""
+                val constraints = inputDescriptorObj["constraints"]!!.jsonObject
+                for (field in constraints["fields"]!!.jsonArray) {
+                    if (field.jsonObject["path"]!!.jsonArray[0].toString() == "\"\$.vct\"") {
+                        val vctField = field.jsonObject
+                        val filter = vctField["filter"]!!.jsonObject
+                        vct = filter["const"]!!.toString().run { substring(1, this.length - 1) }
+                    }
+                }
+                vct
+            } catch (e: Exception) {
```

Comment:
This is a pretty broad catch, which is usually discouraged. Do we know what set of exceptions we're expecting (NPD? anything else?), and can we narrow down this catch a bit?

And would it make sense to add a log line so we can easily see that an exception was hit and what it was?

Reply:
agree that it's broad - I specifically wanted to catch all the not-null assertions in case any of the fields aren't available, so I modified the exception + added the log line